evennia/contrib/grid/xyzgrid/tests.py

```diff
@@ -1415,3 +1416,85 @@ def test_build(self):
         self.assertTrue(room2a.db.desc.startswith("This is the entrance to"))
         self.assertEqual(room2b.key, "North-west corner of the atrium")
         self.assertTrue(room2b.db.desc.startswith("Sunlight sifts down"))
+
+
+mock_room_callbacks = mock.MagicMock()
+mock_exit_callbacks = mock.MagicMock()
+
+class TestXyzRoom(xyzroom.XYZRoom):
+  def at_object_creation(self):
+    mock_room_callbacks.at_object_creation()
+
+class TestXyzExit(xyzroom.XYZExit):
+  def at_object_creation(self):
+    mock_exit_callbacks.at_object_creation()
+
+MAP_DATA = {
+  "map": """
+
+    + 0 1
+
+    0 #-#
+
+    + 0 1
+
+  """,
+  "zcoord": "map1",
+  "prototypes": {
+     ("*", "*"): {
+       "key": "room",
+       "desc": "A room.",
+       "prototype_parent": "xyz_room",
+     },
+     ("*", "*", "*"): {
+       "desc": "A passage.",
+       "prototype_parent": "xyz_exit",
+     }
+  },
+  "options": {
+    "map_visual_range": 1,
+    "map_mode": "scan",
+  }
+}
+
+class TestCallbacks(BaseEvenniaTest):
+    def setUp(self):
+        super().setUp()
+        mock_room_callbacks.reset_mock()
+        mock_exit_callbacks.reset_mock()
+        
+    def setup_grid(self, map_data):
+        self.grid, err = xyzgrid.XYZGrid.create("testgrid")
+
+        def _log(msg):
+             print(msg)
+        self.grid.log = _log
+
+        self.map_data = map_data
+        self.grid.add_maps(map_data)
+
+    def tearDown(self):
+        super().tearDown()
+        self.grid.delete()
+
+    # @override_settings(
```

Comment:
I could not get this work in this case. I can remove it but it would be good to know why.